src/NuGet.Core/NuGet.Commands/PackCommandRunner.cs

```diff
@@ -215,58 +220,44 @@ private static void LoadProjectJsonFile(PackageBuilder builder, string path, str
-                    }
-                    else
-                    {
-                        dependencies.Add(new PackageDependency(dependency.Name, dependency.LibraryRange.VersionRange, flags.Split(new char[] { ',', ' ' }, StringSplitOptions.RemoveEmptyEntries), new List<string>()));
-                    }
-                }
-                PackageDependencyGroup group = new PackageDependencyGroup(NuGetFramework.AnyFramework, dependencies);
-
-                builder.DependencyGroups.Add(group);
+                AddDependencyGroups(spec.Dependencies, NuGetFramework.AnyFramework, builder);
             }
 
             if (spec.TargetFrameworks.Any())
             {
                 foreach (var framework in spec.TargetFrameworks)
                 {
-                    List<PackageDependency> dependencies = new List<PackageDependency>();
-                    foreach (var dependency in framework.Dependencies)
-                    {
-                        if (dependency.IncludeType == LibraryIncludeFlags.None)
-                        {
-                            continue;
-                        }
-
-                        var flags = dependency.IncludeType.ToString();
-                        IReadOnlyList<string> includes = flags.Split(new char[] { ',', ' ' }, StringSplitOptions.RemoveEmptyEntries);
-                        if (includes.SequenceEqual(allIncludeFlags))
-                        {
-                            dependencies.Add(new PackageDependency(dependency.Name, dependency.LibraryRange.VersionRange));
-                        }
-                        else
-                        {
-                            dependencies.Add(new PackageDependency(dependency.Name, dependency.LibraryRange.VersionRange, flags.Split(new char[] { ',', ' ' }, StringSplitOptions.RemoveEmptyEntries), new List<string>()));
-                        }
-                    }
-                    PackageDependencyGroup group = new PackageDependencyGroup(framework.FrameworkName, dependencies);
+                    AddDependencyGroups(framework.Dependencies, framework.FrameworkName, builder);
+                }
+            }
+        }
+
+        private static void AddDependencyGroups(IList<LibraryDependency> dependencies, NuGetFramework framework, PackageBuilder builder)
+        {
+            List<PackageDependency> packageDependencies = new List<PackageDependency>();
+            foreach (var dependency in dependencies)
+            {
+                if (dependency.IncludeType == LibraryIncludeFlags.None)
+                {
+                    continue;
+                }
 
-                    builder.DependencyGroups.Add(group);
+                var flags = dependency.IncludeType.ToString();
+                IReadOnlyList<string> includes = flags.Split(new char[] { ',', ' ' }, StringSplitOptions.RemoveEmptyEntries);
+                if (dependency.IncludeType == LibraryIncludeFlags.All || includes.SequenceEqual(allIncludeFlags))
```

Comment:
Matching NoContent would mean there are no flags since that is the default
If it matches all it needs include="all"

This is very optimized for the defaults I guess so the extra work all ends up on pack